book-a-reading-room-visit.web/Helper/CustomEmailValidation.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Web.Mvc;
+
+namespace book_a_reading_room_visit.web.Helper
+{
+    public class EmailAttribute
+        : RegularExpressionAttribute, IClientValidatable
+    {
+        public EmailAttribute()
+            //updated to llow new domain names in the email addresses, such as james@jamesharrison.productions
+            //by increasing the length of characters up to 24
+            : base(@"^([a-zA-Z0-9_\-\+\'\.]+)@((\[[0-9]{1,3}" +
+            @"\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\" +
+            @".)+))([a-zA-Z]{2,24}|[0-9]{1,3})(\]?)$")
```

Comment:
@Puththiran I see in KBS there is an email with domain @whoamifamilytreeresearch.co.uk which is 30 characters long. And I see in ASPIdentity there is @fh05yolxcxllqsv6rqomy696vx1opd.burpcollaborator.net which is 51 characters long.
Is this regex working? It should restrict the len to 24.